src/machine-learning/machine-learning-api-client.ts

```diff
@@ -37,17 +37,28 @@ export interface StatusErrorResponse {
 }
 
 /**
- * A Firebase ML Model input object
+ * Firebase ML Model input objects
  */
-export interface ModelOptions {
+export interface ModelOptionsBase {
   displayName?: string;
   tags?: string[];
-
-  tfliteModel?: { gcsTfliteUri: string };
 }
+export interface GcsTfliteModelOptions extends ModelOptionsBase {
+  tfliteModel: {
+    gcsTfliteUri: string;
+  };
+}
+export interface AutoMLTfliteModelOptions extends ModelOptionsBase {
+  tfliteModel: {
+    automlModel: string;
+  };
+}
+export type ModelOptions = ModelOptionsBase | GcsTfliteModelOptions | AutoMLTfliteModelOptions;
+export type ModelUpdateOptions = ModelOptions & { state?: { published?: boolean }};
```

Comment:
```
export interface ModelUpdateOptions extends ModelOptions {
  state?: { published?: boolean }
}
```

Reply:
Gives me an error: "An interface can only extend an object type or intersection of object types with statically known members."

Reply:
Turns out you cannot extend union types. But we can avoid this if we were to just drop the union type as described in my comment above.

Reply:
What would you extend? 
If ModelUpdateOptions is just an extension of the current ModelOptionsBase, then what about GcsTFLiteModelOptions? Would we need to do another definition of GcsTFLiteModelUpdateOptions that extends ModelUpdateOptions? (and the same for every new model options object we define (there will likely be many in future)). I prefer the current implementation.